beard/clustering/blocking_funcs.py

```diff
@@ -0,0 +1,487 @@
+    # times.
+    FULL_NAME_MATCH_BONUS = 10
+
+    def first_names_score(self, new_first_names, last_name):
+        """Count matches among the initials.
+
+        A match is defined as match of all the first names from the new
+        signature with the some of the names from the old signature.
+
+        Such a match needs to keep order. For example ("A", "S") will match
+        ("A", "D", "S"), but won't match ("S", "A", "D").
+
+        In case of full first names available the match considers:
+        + Full name match if both compared element are full names. Such a match
+        is counted FULL_NAME_MATCH_BONUS times to the result, as it indicates
+        the same author with bigger probability.
+        + Initials match if none of compared elements is a full name.
+        + Initials match if one od the compared elements is a full name.
+
+        Two characters in initials - ``A`` and ``H`` are handled in special
+        way in initial comparison, as the double metaphone can output an empty
+        string for them.
+
+        Parameters
+        ----------
+        :param new_first_names: tuple
+            Strings representing given names of the new author
+        :param last_name: tuple
+            Tokens, usually one, representing last surname(s) of the author.
+
+        Raises
+        ------
+        :raises: KeyError
+            When the last name is not included in the cluster
+
+        Returns
+        -------
+        :returns: integer
+            Score for matching first names in the cluster.
+        """
+        result = 0
+        if last_name in self._content:
+            for first_names, occurences in \
+                    six.iteritems(self._content[last_name]):
+                first_names_length = len(first_names)
+                old_names_index = 0
+                names_match = self.FULL_NAME_MATCH_BONUS
+                for new_name in new_first_names:
+                    while old_names_index < first_names_length:
+                        full_names_match = \
+                            self._names_match(first_names[old_names_index],
+                                              new_name)
+                        if not full_names_match:
+                            # There was no match, check next name from the
+                            # old signature
+                            old_names_index += 1
+                        else:
+                            if names_match:
+                                names_match = full_names_match
+                            # Go to the next name from the new ones.
+                            break
+                    if old_names_index == first_names_length:
+                        # No match.
+                        names_match = 0
+                        break
+                if names_match:
+                    result += occurences * names_match
+            return result
```

Comment:
This algorithm gives equal scores matching for examples 'Mark' to 'Mark' and to ' Mark Bell'. I propose dividing by the number of tokens to have better scoring